src/main/java/org/benetech/servicenet/service/impl/OrganizationMatchServiceImpl.java

```diff
@@ -218,15 +240,27 @@ public void createOrUpdateOrganizationMatches(Organization organization, Matchin
                 .map(m -> m.getPartnerVersion().getId())
                 .collect(Collectors.toList());
 
-            List<Organization> notMatchedOrgs = findNotMatchedOrgs(currentMatchesIds, organization.getAccount().getName());
+            List<Organization> partnerOrganizations = findOrganizationsExcept(organization.getAccount().getName());
 
-            findAndPersistMatches(organization, notMatchedOrgs, context);
+            List<OrganizationMatch> matches = findAndPersistMatches(organization, partnerOrganizations, context);
+            removeObsoleteMatches(matches, currentMatches);
```

Comment:
would it make more sense to rename currentMatches -> previousMatches? It looks like it is used that way in removeObsoleteMatches method.

Reply:
they become 'previousMatches' just after findAndPersistMatches is called.